pkgs/development/compilers/scala/dotty.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchurl, makeWrapper, jre }:
+
+stdenv.mkDerivation rec {
+  version = "0.4.0-RC1";
+  name = "dotty-${version}";
+
+  src = fetchurl {
+    url = "https://github.com/lampepfl/dotty/releases/download/${version}/${name}.tar.gz";
+    sha256 = "1d1ab08b85bd6898ce6273fa50818de0d314fc6e5377fb6ee05494827043321b";
+  };
+
+  propagatedBuildInputs = [ jre ] ;
+  buildInputs = [ makeWrapper ] ;
+
+  installPhase = ''
+    mkdir -p $out
+    mv * $out
+
+    for p in $out/bin/* ; do
+      file=$(basename $p)
+
+      # no need to wrap common
+      if [[ "$file" = "common" ]] ; then
+        continue
+      fi
+
+      wrapProgram $p --set JAVA_HOME ${jre}
+    done    
+  '';
+
+  meta = {
```

Comment:
Just a note: you can skip `stdenv.lib` inside `meta` if you add `with stdenv.lib;` to the attribute set:

```
meta = with stdenv.lib; {
  license = licenses.bsd3;
  platforms = platforms.all;
};
```

Reply:
Ok